### PR TITLE
Fix detecting arhitecture on raspberry pi

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Detect architecture depending on size of void

Related to #1329

### CODE REVIEW

File: tools/cmake/helpers/platform.cmake [hunk 1]
```diff
@@ -74,20 +74,16 @@ function(get_os OS)
...
 function(get_arch ARCH)
-  if(ARCH_X86)
+  if( CMAKE_SIZEOF_VOID_P MATCHES 8 )
+    # void ptr = 8 byte --> x86_64
+    set(${ARCH} "x86_64" PARENT_SCOPE) 
+  elseif( CMAKE_SIZEOF_VOID_P MATCHES 4 )
+    # void ptr = 4 byte --> x86
     set(${ARCH} "x86" PARENT_SCOPE)
-  elseif(ARCH_X64)
-    set(${ARCH} "x64" PARENT_SCOPE)
```
Comment (jacobkeeler) [inline tools/cmake/helpers/platform.cmake]: Do you know whether the change from `x64` to `x86_64` will break anything? From what I've seen of the CMake changes, it looks like this `ARCH` value is only really relevant when building for Windows.

File: tools/cmake/helpers/platform.cmake [hunk 2]
```diff
@@ -74,20 +74,16 @@ function(get_os OS)
 endfunction()
 
 function(get_arch ARCH)
-  if(ARCH_X86)
+  if( CMAKE_SIZEOF_VOID_P MATCHES 8 )
+    # void ptr = 8 byte --> x86_64
+    set(${ARCH} "x86" PARENT_SCOPE)
```
Comment (jacobkeeler) [inline tools/cmake/helpers/platform.cmake]: Typo, should be x64

**Comment by LuxoftAKutsan** [comment]
@mrapitis grate! @jacobkeeler @JackLivio please review and merge

**Comment by jacobkeeler** [comment]
Ran into the same "unsupported architecture" message when rebasing #1159 onto the new develop, incorporating this fix resolved the issue.

**Pull request merged**